src/xmpp/iq.c

```diff
@@ -158,6 +158,7 @@ static int
 _iq_handler(xmpp_conn_t *const conn, xmpp_stanza_t *const stanza, void *const userdata)
 {
     log_debug("iq stanza handler fired");
+    iq_autoping_timer_cancel();
```

Comment:
Why do you cancel timer on every incoming IQ stanza? When does it suppose to start again?

Reply:
Every IQ you receive proofs you're connected to the server so they are as good as ping result I guess.

Reply:
I misunderstood the timer semantics. I thought it is always the same timer, but it rather starts on every ping.

I won't insist on waiting for only ping reply. But consider the following when you decide what to push to master: receiving a stanza after sending a ping doesn't mean that the stanza arrived after ping had been sent. The stanza could reside in kernel's buffers. And in this scenario you will need 1 more ping interval to detect broken connection. This is not a big deal for profanity's usecases, so it is up to you to decide how to proceed.